### ISSUE TITLE
Include AWX (or other UI) in Ansible appliance

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#1366]
Mike (via the [forums](https://www.turnkeylinux.org/forum/general/wed-20191023-1346/ansible-awx)) has requested we include [AWX](https://github.com/ansible/awx) in our [Ansible appliance](https://www.turnkeylinux.org/ansible).

AWX is the open source (Apache licensed) project that underlies [Ansible Tower](https://www.ansible.com/products/tower). AWX provides a WebUI, REST API, and task engine for Ansible.

I think it would be a fantastic inclusion! :smile: Install instructions are [here](https://github.com/ansible/awx/blob/devel/INSTALL.md).

**Comment by JedMeister** [comment]
This has been deferred for now... It would be great to add it, but it's blocked by the fact that it's currently only installable via docker.

It's probably possible to install a "dev environment" from source, but I anticipate that will be a burden for users to update, so we'll need to think a bit more about this...

A significant amount of software seems to be distributed via docker these days, so my thought is that we should probably look to support usage of docker within TKLDev builds - i.e. #1502.

**Comment by JedMeister** [comment]
Whilst in theory, this would be cool, the current state of affairs suggest that it may not be a viable option.

Firstly, from a purely pragmatic perspective, the software is currently only being distributed via Docker. So until we can pre-package docker containers, using the upstream recommended install path is blocked.

Also, on closer reading of the docs, plus reports via other online sources suggest that "AWX" is essentially just the "dev" (or perhaps "beta") branch of "Ansible Tower". So we'd essentially be beta testing a future Tower release. Many claim that it is often buggy and not at all stable (primarily in the sense of API stability; although some have also critiqued the performance stability too).

I was considering closing this for v16.1 and tagging with "won't fix". However, then I came across [Semaphore](https://ansible-semaphore.com/). So we should include that in the Ansible appliance for the v16.1 release.

I'm going to close this (once Semaphore is included). If/when we have support for inclusion of docker containers and/or there is an alternate way to install AWX then we could reconsider. Although assuming that the feedback is positive, I'm inclined to stick with Semaphore.
